File: mini-program/src/api/carousel.ts
// 轮播图相关接口
import { http } from './request'

export interface CarouselImageItem {
  id: number
  image_url: string
  title: string
  description: string
  sort_order: number
  created_at: string
}

export function getCarouselImages() {
  return http.get<CarouselImageItem[]>('/carousel')
}

File: server/app/schemas/carousel.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CarouselImageOut(BaseModel):
    """公开轮播图输出。"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    image_url: str
    title: str = ""
    description: str = ""
    sort_order: int = 0
    created_at: datetime